cmake/modules/DetectLoongArchFeatureASX.c

```diff
@@ -0,0 +1,7 @@
+#include <lasxintrin.h>
+
+int main(void) {
+  __m256i a, b, c;
+  c = __lasx_xvand_v(a, b);
+  return 0;
+}
```

Comment:
How about renaming this file to `SimpleProgramCallLSXBuiltin.c` which is more semantically consistent?

Reply:
Another alternative is writing this temporary file when running cmake. We can refer the `try_run` usage in llvm-project: `llvm-project/llvm/cmake/modules/FindZ3.cmake`.

The pros is we can keep the `cmake/modules` directory as clean as possible.